### ISSUE TITLE
smart-create

### ISSUE THREAD

**Issue by phunehehe** [zalora/myrapi#7]
(Follow up from #4)

I think this will be a good interface for `smart-create`:

``` bash
myrapi smart-create --subdomain abc.example.com --type CNAME --ttl 300 \
    --values "abc1.example.com abc2.example.com abc3.example.com"
```

For each value, the command should check if a matching record exists (based on subdomain and value) and call `create` or `update` accordingly.

After everything is done, the command should check the number of records for that subdomain versus the number of values given, and emits a warning if they don't match.

**Comment by jekor** [comment]
> myrapi smart-create --subdomain abc.example.com --type CNAME --ttl 300 --values "abc1.example.com abc2.example.com abc3.example.com"

When does having multiple CNAME records make sense? If we're going to do smart creation, we might want to deal with it with some higher-level operations that express our intent. There are only 2 cases I can think of where we want to create multiple records. One is for load balancing, where the multiple A records are just an artifact of their interface:

```
myrapi load-balance --domain www.zalora.com.my ip1 ip2 ip3
```

The other is for text records, which I doubt we'll make automate any time soon. For doing the correct thing for cnames, we could have a similar high-level command:

```
myrapi alias --domain static1.zalora.com.my dn8woun8q4kj1.cloudfront.net
```

**Comment by phunehehe** [comment]
That's right. I was thinking of `smart-create` as a high level operation, but I didn't think of giving it a good name. `load-balance` indeed sounds better.
